app/src/main/java/com/ablylabs/pubcrawler/realtime/FlowyRealtimePub.kt

```diff
@@ -0,0 +1,103 @@
+package com.ablylabs.pubcrawler.realtime
+
+import android.util.Log
+import com.ablylabs.pubcrawler.pubs.Pub
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+private const val TAG = "FlowyRealtimePub"
+
+interface FlowyRealtimePub {
+    suspend fun join(pub: Pub, who: PubGoer): JoinResult
+    suspend fun leave(who: PubGoer, which: Pub): LeaveResult
+    suspend fun buildActionFlow(pub: Pub, who: PubGoer): Flow<PubActions>
+    suspend fun buildPresenceFlow(pub: Pub): Flow<PubPresenceActions>
+    suspend fun sendTextMessage(
+        who: PubGoer,
+        toWhom: PubGoer,
+        messageText: String
+    ): MessageSentResult
+
+    suspend fun acceptDrink(who: PubGoer, fromWhom: PubGoer): AcceptDrinkResult
+    suspend fun rejectDrink(who: PubGoer, fromWhom: PubGoer): RejectDrinkResult
+    suspend fun offerDrink(who: PubGoer, toWhom: PubGoer): OfferSentResult
+    suspend fun registerToDrinkOfferResponse(offered: PubGoer, offeree: PubGoer): DrinkOfferResponse
+
+    //all pubgoers in a pub
+    suspend fun allPubGoers(which: Pub): List<PubGoer>
+}
+
+class FlowyPubImpl(private val suspendyPub: SuspendyRealtimePub) : FlowyRealtimePub {
```

Comment:
Why do you need this `FlowyRealtimePub`? Isn't `SuspendyRealtimePub` enough? :thinking:

Reply:
It's not really needed but I wanted to create a class that will perform some combined operations for UI (such as buildActionFlow) In the end I also questioned the existence of this but as I was experimenting on it, why not 😆